### PR TITLE
OSDOCS-318: Included introduction to Persistent Storage.

### PR DESCRIPTION
**Pull request by huffmanca**
OSDOCS-318: Included introduction to Persistent Storage.

This is for OS 4.0.

### CODE REVIEW

**Comment by huffmanca** [comment]
@openshift/team-documentation , is it possible for someone to peer review this?

File: modules/storage-persistent-storage-pv.adoc [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+ *  *emptyDir* has the same lifecycle as the pod:
+   ** *emptyDir* volumes survive container crashes/restarts.
+   ** *emptyDir* volumes are deleted when the pod is deleted.
+====
+endif::[]
+
+[[pv-reclaim-policy]]
+== Reclaim policy
+
+The following table lists current reclaim policies:
```
Comment (bergerhoffer) [inline modules/storage-persistent-storage-pv.adoc]: There is only one policy, should this just say policy? (same with the table title below)

File: modules/storage-persistent-storage-pv.adoc [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+|Released
+|The claim was deleted, but the resource is not yet reclaimed by the
+cluster.
+
+|Failed
+|The volume has failed its automatic reclamation.
+
+|===
+
+The CLI shows the name of the PVC bound to the PV.
```
Comment (bergerhoffer) [inline modules/storage-persistent-storage-pv.adoc]: This sentence seems weird to me. Should it say something like "You can see the name of the PVC bound to the PV using the following CLI command:" and then list the CLI command?

File: modules/storage-persistent-storage-pvc.adoc [hunk 3]
```diff
@@ -0,0 +1,91 @@
+// Module included in the following assemblies:
+//
+// * storage/understanding-persistent-storage.adoc
```
Comment (bergerhoffer) [inline modules/storage-persistent-storage-pvc.adoc]: I don't see this file included in this or any assembly, should it be?

**Review by bergerhoffer — COMMENTED** [review]
A few minor things, but overall looks really great @huffmanca, nice job!

File: _topic_map.yml [hunk 4]
```diff
@@ -293,6 +293,20 @@ Topics:
   File: optimizing-storage
 - Name: Routing optimization
   File: routing-optimization
+#---
```
[the remark this answers is not in the document]
Comment (huffmanca) [reply _topic_map.yml]: I've removed the comments on this file for the storage section.

File: modules/storage-persistent-storage-block-volume-examples.adoc [hunk 5]
```diff
@@ -0,0 +1,142 @@
+// Module included in the following assemblies:
+//
+// * storage/understanding-persistent-storage.adoc
+//
+// This module should only be present in openshift-enterprise and
+// openshift-origin distributions.
+
+ifdef::openshift-enterprise,openshift-origin[]
+[id='block-volume-examples-{context}']
+= Block Volume Examples
```
[the remark this answers is not in the document]
Comment (huffmanca) [reply modules/storage-persistent-storage-block-volume-examples.adoc]: Adjusted to sentence case.

File: modules/storage-persistent-storage-lifecycle.adoc [hunk 6]
```diff
@@ -0,0 +1,90 @@
+// Module included in the following assemblies:
+//
+// * storage/understanding-persistent-storage.adoc
+
+[id=lifecycle-volume-claim-{context}]
+= Lifecycle of a Volume and Claim
```
[the remark this answers is not in the document]
Comment (huffmanca) [reply modules/storage-persistent-storage-lifecycle.adoc]: Adjusted to sentence case.

File: modules/storage-persistent-storage-block-volume-examples.adoc [hunk 7]
```diff
@@ -0,0 +1,142 @@
+// Module included in the following assemblies:
+//
+// * storage/understanding-persistent-storage.adoc
+//
+// This module should only be present in openshift-enterprise and
+// openshift-origin distributions.
+
+ifdef::openshift-enterprise,openshift-origin[]
```
[the remark this answers is not in the document]
Comment (huffmanca) [reply modules/storage-persistent-storage-block-volume-examples.adoc]: I've moved these conditionals into the assembly and left the comment in each module.

File: modules/storage-persistent-storage-pv.adoc [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+== Restrictions
+
+The following restrictions apply when using PVs with {product-title}:
+endif::[]
+
+ifdef::openshift-dedicated[]
+[IMPORTANT]
+====
+ * PVs are provisioned with either EBS volumes (AWS) or GCP storage (GCP), 
+depen ding on where the cluster is provisioned.
```
[the remark this answers is not in the document]
Comment (huffmanca) [reply modules/storage-persistent-storage-pv.adoc]: Corrected typos.

File: modules/storage-persistent-storage-pv.adoc [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+
+ifdef::openshift-dedicated,openshift-online[]
+[[pv-restrictions]]
+== Restrictions
+
+The following restrictions apply when using PVs with {product-title}:
+endif::[]
+
+ifdef::openshift-dedicated[]
+[IMPORTANT]
```
[the remark this answers is not in the document]
Comment (huffmanca) [reply modules/storage-persistent-storage-pv.adoc]: I've removed the IMPORTANT blocks around both sections. This now leads into the bulleted list directly.

File: modules/storage-persistent-storage-pvc.adoc (hunk 3 above)
Comment (huffmanca) [reply modules/storage-persistent-storage-pvc.adoc]: It should! I've included it immediately below the PV module.

File: _topic_map.yml [hunk 10]
```diff
@@ -293,6 +293,20 @@ Topics:
...
+#Name: Storage
+#Dir: storage
+#Distros: openshift-*
+#Topics:
+#- Name: Persistent Storage
```
[the remark this answers is not in the document]
Comment (huffmanca) [reply _topic_map.yml]: I've adjusted the entries in this file to use sentence case.

File: modules/storage-persistent-storage-pv.adoc (hunk 1 above)
Comment (huffmanca) [reply modules/storage-persistent-storage-pv.adoc]: I've adjusted this to be singular for now. If additional policies are included I'll change it back.

File: modules/storage-persistent-storage-pv.adoc (hunk 2 above)
Comment (huffmanca) [reply modules/storage-persistent-storage-pv.adoc]: I don't have an active cluster right now. I've filed https://jira.coreos.com/browse/OSDOCS-332 to follow up on this once one is available. Marking resolved for now, and will revisit this issue under the above JIRA.

**Comment by huffmanca** [comment]
@bergerhoffer ,

Thank you for the review! Merging this one.

**Comment by bergerhoffer** [comment]
Hey @huffmanca, just wanted to make sure you saw all of my comments. Because you replied to all of them except the ones where it said they were hidden and you had to hit "Load more" in the Conversation tab to see them. But you should be able to see them in the diff tab. Anyways, there are a few that might need to get followed up on, like a "fornode" -> "for node".

**Comment by huffmanca** [comment]
@bergerhoffer ,

You're correct - I missed those due to them being collapsed. Apologies for this oversight. The rest of these changes have been submitted in https://github.com/openshift/openshift-docs/pull/13994 , and I've assigned you as the reviewer for this PR as well.

**Pull request merged**